### ISSUE TITLE
Reduce time for XDMF tests

### ISSUE THREAD

**Issue by garth-wells** [FEniCS/dolfinx#13]
Tests in `test.XDMF.py` are way too slow. Python loops over entities of non-trivial meshes are performed.

**Comment by garth-wells** [comment]
@michalhabera ?

**Comment by michalhabera** [comment]
I bloated it adding parameters and combinations for checkpoint methods. Parameters are enumerated in the beginning of file so its matter of changing few lines. 

I also started writing completely new tests, but the idea is to make it more systematic, not the slow execution. https://bitbucket.org/michalhabera/fenics-io-tests/src/ac9a820bd174720865975468f626d793476c975b/python-tests/test_XDMF.py?at=master&fileviewer=file-view-default

Are there long running tests somewhere stored and run? If you want to have just fast ones then I'll reduce it tomorrow.
